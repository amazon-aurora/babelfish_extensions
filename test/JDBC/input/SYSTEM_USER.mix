-- tsql
SELECT session_user, system_user, current_user, db_name();
GO

CREATE LOGIN r1 WITH PASSWORD = '123';
GO

CREATE LOGIN r2 WITH PASSWORD = '123';
GO

-- tsql user=r1 password=123
SELECT session_user, system_user, current_user, db_name();
GO

-- tsql user=r2 password=123
SELECT session_user, system_user, current_user, db_name();
GO


-- psql
-- Need to terminate active session before cleaning up the login
SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL) 
WHERE sys.suser_name(usesysid) in ('r1','r2') AND backend_type = 'client backend' AND usesysid IS NOT NULL;
GO

-- tsql
DROP LOGIN r1;
DROP LOGIN r2;
GO