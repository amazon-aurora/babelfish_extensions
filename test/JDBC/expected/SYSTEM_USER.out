-- tsql
SELECT session_user, system_user, current_user, db_name();
GO
~~START~~
varchar#!#nvarchar#!#varchar#!#nvarchar
jdbc_user#!#jdbc_user#!#dbo#!#master
~~END~~


CREATE LOGIN r1 WITH PASSWORD = '123';
GO

CREATE LOGIN r2 WITH PASSWORD = '123';
GO

-- tsql user=r1 password=123
SELECT session_user, system_user, current_user, db_name();
GO
~~START~~
varchar#!#nvarchar#!#varchar#!#nvarchar
r1#!#r1#!#guest#!#master
~~END~~


-- tsql user=r2 password=123
SELECT session_user, system_user, current_user, db_name();
GO
~~START~~
varchar#!#nvarchar#!#varchar#!#nvarchar
r2#!#r2#!#guest#!#master
~~END~~



-- psql
-- Need to terminate active session before cleaning up the login
SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL) 
WHERE sys.suser_name(usesysid) in ('r1','r2') AND backend_type = 'client backend' AND usesysid IS NOT NULL;
GO
~~START~~
bool
t
t
~~END~~

-- Wait to sync with another session
SELECT pg_sleep(1);
GO
~~START~~
void

~~END~~


-- tsql
DROP LOGIN r1;
DROP LOGIN r2;
GO
